3-java-servlet-web-app/1-Authentication/sign-in-ciam/README.md

```diff
@@ -0,0 +1,272 @@
+
+#### Deploying the Sample
+
+Our samples can be deployed to a number of application servers, such as Tomcat, WebLogic, or Webshpere, and MSAL Java itself can generally be integrated into existing applications without changes to your existing deployment set up.
+
+You can find instructions for deploying our samples [here on MSAL Java's Github wiki](https://github.com/AzureAD/microsoft-authentication-library-for-java/wiki/Deployment-Instructions-for-MSAL-Java-Samples).
+
+## Explore the sample
+
+- Note the signed-in or signed-out status displayed at the center of the screen.
+- Click the context-sensitive button at the top right (it will read `Sign In` on first run)
+- Follow the instructions on the next page to sign in with an account in the Microsoft Entra tenant.
+- On the consent screen, note the scopes that are being requested.
+- Note the context-sensitive button now says `Sign out` and displays your username to its left.
+- The middle of the screen now has an option to click for **ID Token Details**: click it to see some of the ID token's decoded claims.
+- You can also use the button on the top right to sign out.
+- After signing out, click the link to `ID Token Details` to observe that the app displays a `401: unauthorized` error instead of the ID token claims when the user is not authorized.
+
+> :information_source: Did the sample not work for you as expected? Did you encounter issues trying this sample? Then please reach out to us using the [GitHub Issues](../../../../issues) page.
+
+## We'd love your feedback!
+
+Were we successful in addressing your learning objective? Consider taking a moment to [share your experience with us](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR73pcsbpbxNJuZCMKN0lURpURDQwVUxQWENUMlpLUlA0QzdJNVE3TUJRSyQlQCN0PWcu).
+
+## About the code
+
+This sample shows how to use **MSAL for Java (MSAL4J)** to sign in users into your Microsoft Entra tenant. If you'd like to use MSAL4J in your own applications, must add it to your projects using Maven.
+
+If you want to replicate this sample's behavior, you may choose to copy the `pom.xml` file, and the contents of the `helpers` and `authservlets` packages in the `src/main/java/com/microsoft/azuresamples/msal4j` package. You'll also need the [authentication.properties file](src/main/resources/authentication.properties). These classes and files contain generic code that can be used in a wide array of applications. The rest of the sample may be copied as well, but the other classes and files are built specifically to address this sample's objective.
+
+A **ConfidentialClientApplication** instance is created in the [AuthHelper.java](src/main/java/com/microsoft/azuresamples/authentication/AuthHelper.java) class. This object helps craft the Microsoft Entra authorization URL and also helps exchange the authentication token for an access token.
+
+```Java
+// getConfidentialClientInstance method
```

Comment:
dead code ?

Reply:
This was just a comment saying that the code is a snippet of the `getConfidentialClientInstance` method. The method itself wasn't much longer than the snippet, so in the latest commit I just added the whole method to this section.